### DISCUSSION TITLE
Diarization for Italian language keeps giving me "speaker-0" to anybody's talking

### DISCUSSION THREAD

**Discussion by M-C-Dev** [deepgram/community#838] [General help]
### Which Deepgram product feature are you using?

Deepgram API - STT Pre-recorded

### Details

I'm working at the moment on diarization, trying to evaluate DeepGram's results. 
For now, I'm just keeping testing via simple curls.

The answer is fast, and the text is transcripted very precisely, but diarization's lacks a lot.

Tried also put "nova-2-meeting" or "nova-2-phonecall" like advised here: [https://github.com/orgs/deepgram/discussions/584](url) but i got 400 response.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?diarize=true&language=it&model=nova-2

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

6a4c7485-b243-4d87-b3c5-c483fcaa8d8b

### If possible, please attach your code or paste it into the text box.

curl --location 'https://api.deepgram.com/v1/listen?diarize=true&language=it&model=nova-2' \
--header 'Authorization: token <>' \
--header 'Content-Type: audio/m4a' \
--data '@/C:/Users/marco/Documents/Registrazioni di suoni/Registrazione in corso (3).m4a'

### If possible, please attach an example audio file to reproduce the issue.

https://file.io/yUD6zlGmYjl4

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment]
@M-C-Dev I tried out your audio and had similar results. The first thing I always do when it's an Mp4 file is convert it to wav or mp3, so I definitely recommend doing that. However, the results weren't great for this sample audio.

Cases of diarization not working are definitely frustrating. Depending on various factors such as length and distribution of each person speaking, and audio properties such as quality and file format, diarization may fail to distinguish among multiple speakers. Diarization is one of the trickiest things to get right for STT models. Deepgram is working hard to improve our diarization, but we still have a lot of work to do.

My first recommendation is to make sure you use longer samples. Diarization works better with longer audio files. If you have a short audio file, enabling diarization can result in lower accuracy if there isn't enough audio data for the model to effectively differentiate between speakers.  So that is probably the case with this file. Would probably need approximately 30 seconds per speaker for it to work.

Another recommendation I give in this situation, in cases where diarization is crucial to the use case, is to transcribe [multi-channel audio](https://developers.deepgram.com/docs/multichannel). This is where two speakers on the audio are each on their own audio channel. That enables perfect speaker segmentation as each speaker's audio is isolated already. This is what our customers do if it is absolutely vital that the separate speakers are transcribed with their own speaker labels.
